spec/services/courses/copy_to_provider_service_spec.rb

```diff
@@ -35,34 +42,28 @@
   end
 
   it 'leaves the existing course alone' do
-    service.execute(new_provider)
+    service.execute(course: course, new_provider: new_provider)
 
     expect(provider.reload.courses).to eq [course]
   end
 
-  context 'course has a published but no draft enrichment' do
-    let!(:published_enrichment) do
-      create :course_enrichment, :published, course: course
-    end
-
-    before { service.execute(new_provider) }
-
-    subject { new_course.enrichments }
-
-    its(:length) { should eq 1 }
+  it "doesn't copy enrichments when they do not exist" do
+    service.execute(course: course, new_provider: new_provider)
 
-    describe 'the new course' do
-      subject { new_course }
+    expect(mocked_enrichments_copy_to_course_service).to_not have_received(:execute)
+  end
 
-      its(:content_status) { should eq :rolled_over }
+  context 'when a published enrichment exists' do
+    let!(:published_enrichment) do
+      create :course_enrichment, :published, course: course
     end
 
-    describe 'the copied enrichment' do
-      subject { new_course.enrichments.first }
+    it 'copies the published enrichment' do
```

Comment:
Worth also testing that is copies the _latest_ enrichment regardless of status :eyes: